history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval int64) *History {
+	h := new(History)
+	h.tree = m
+	h.epochInterval = epInterval
+	h.currentSTR = newSTR(m, startEp, 0, make([]byte, crypto.HashSizeByte), key)
+	return h
+}
+
+func (h *History) UpdateHistory(m *MerkleTree, nextEp int64) error {
```

Comment:
> But even if we included all TB_0s in a batch at the end of t=0 like you had originally suggested (and as we do in the reference implementation), since the initial epoch t=0 has an STR, the next STR for t=1 and tree_1 would be computed based on all TB_0s.

Is this wrong? According to the paper:
> The binding is signed by the identity provider, creating a non-repudiable promise to add this data to the next version of the tree.

> The client then needs to wait for the next epoch and ensure that the provider has kept its promise of inserting Alice’s binding into its key directory by the next epoch.

So as I understand, the STR of epoch `t` and `tree_t` would be computed based on all `TB_(t-1)`s (i.e., `STR[1]` of `tree_1` should be computed based on all `TB_0`s), no?